tests/infrared/16.1/infrared-openstack.sh

```diff
@@ -0,0 +1,116 @@
+VM_IMAGE="${VM_IMAGE:-rhel-guest-image-8.2-290.x86_64.qcow2}"
+VM_IMAGE_LOCATION="${VM_IMAGE_URL_PATH}/${VM_IMAGE}"
+
+OSP_BUILD="${OSP_BUILD:-passed_phase2}"
+OSP_VERSION="${OSP_VERSION:-16.1}"
+OSP_TOPOLOGY="${OSP_TOPOLOGY:-undercloud:1,controller:3,compute:2,ceph:3}"
+OSP_MIRROR="${OSP_MIRROR:-rdu2}"
+LIBVIRT_DISKPOOL="${LIBVIRT_DISKPOOL:-/var/lib/libvirt/images}"
+
+TEMPEST_ONLY="${TEMPEST_ONLY:-false}"
+
+ir_run_cleanup() {
+  infrared virsh \
+      -vv \
+      -o outputs/cleanup.yml \
+      --disk-pool "${LIBVIRT_DISKPOOL}" \
+      --host-address "${VIRTHOST}" \
+      --host-key "${SSH_KEY}" \
+      --cleanup yes
+}
+
+ir_run_provision() {
+  infrared virsh \
+      -vvv \
+      -o outputs/provision.yml \
+      --disk-pool "${LIBVIRT_DISKPOOL}" \
+      --topology-nodes "${OSP_TOPOLOGY}" \
+      --host-address "${VIRTHOST}" \
+      --host-key "${SSH_KEY}" \
+      --image-url "${VM_IMAGE_LOCATION}" \
+      --host-memory-overcommit True \
+      -e override.controller.cpu=4 \
+      -e override.controller.memory=16384 \
+      --serial-files True
+}
+
+ir_create_undercloud() {
+  infrared tripleo-undercloud \
+      -vv \
+      -o outputs/undercloud-install.yml \
+      --mirror "${OSP_MIRROR}" \
+      --version "${OSP_VERSION}" \
+      --build "${OSP_BUILD}" \
+      --images-task rpm \
+      --images-update no \
+      --tls-ca https://password.corp.redhat.com/RH-IT-Root-CA.crt \
+      --config-options DEFAULT.undercloud_timezone=UTC
+}
+
+stf_create_config() {
+  sed -e "s/<<AMQP_HOST>>/${AMQP_HOST}/;s/<<AMQP_PORT>>/${AMQP_PORT}/" stf-connectors.yaml.template > outputs/stf-connectors.yaml
+}
+
+ir_create_overcloud() {
+  infrared tripleo-overcloud \
+      -vv \
+      -o outputs/overcloud-install.yml \
+      --version "${OSP_VERSION}" \
+      --deployment-files virt \
+      --overcloud-debug yes \
+      --network-backend geneve \
+      --network-protocol ipv4 \
+      --network-dvr yes \
+      --storage-backend ceph \
+      --storage-external no \
+      --overcloud-ssl no \
+      --introspect yes \
+      --tagging yes \
+      --deploy yes \
+      --ntp-server "${NTP_SERVER}" \
+      --overcloud-templates outputs/stf-connectors.yaml \
+      --containers yes
+}
+
+ir_run_tempest() {
+  infrared tempest \
+      -vv \
+      -o outputs/test.yml \
+      --openstack-installer tripleo \
+      --openstack-version "${OSP_VERSION}" \
+      --tests smoke \
+      --setup rpm \
+      --revision=HEAD \
+      --image http://download.cirros-cloud.net/0.4.0/cirros-0.4.0-x86_64-disk.img
+}
+
+ir_expose_ui() {
+  infrared cloud-config --deployment-files virt --tasks create_external_network,forward_overcloud_dashboard,launch_workload
+}
+
+if ${TEMPEST_ONLY}; then
+  echo "-- Running tempest tests"
+  ir_run_tempest
+else
+  echo "-- full cloud deployment"
+  ir_run_cleanup
+  ir_run_provision
+  ir_create_undercloud
+  stf_create_config
+  ir_create_overcloud
+  # ir_expose_ui
```

Comment:
Break this out to a switch that can be enabled if you're going to comment it out